### ISSUE TITLE
Label column 'Label' not found (Parameter 'schema')

### ISSUE THREAD

**Issue by kamranalam27** [dotnet/machinelearning-samples#783]
using FypModule1.Models;
using Microsoft.AspNetCore.Mvc;
using Microsoft.ML;
using Microsoft.ML.Data;
using Microsoft.ML.Trainers;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using Microsoft.ML.Transforms;
using System.Collections;
using System.Data;
using System.Data.Common;
using System.Data.SqlClient;
using System.Diagnostics;
using System.Drawing;
using System.Net;
using Microsoft.EntityFrameworkCore;

namespace FypModule1.Controllers
{
    public class ProductController : Controller
    {
        public static IEnumerable<string> ReadRemoteDataset(string url)
        {
            using (var client = new WebClient())
            using (var stream = client.OpenRead(url))
            using (var reader = new StreamReader(stream))
            {
                string line;
                while ((line = reader.ReadLine()) != null)
                {
                    yield return line;
                }
            }
        }

        /// <summary>
        /// Wrapper function that performs the database query and returns an IEnumerable, creating
        /// a database context each time.
        /// </summary>
        /// <remarks>
        /// ML.Net can traverse an IEnumerable with multiple threads. This will result in Entity Core Framwork throwing an exception
        /// as multiple threads cannot access the same database context. To work around this, create a database context
        /// each time a IEnumerable is requested.
        /// </remarks>
        /// <returns>An IEnumerable of the resulting data.</returns>
        private static IEnumerable<FrequentlyBoughtTogetherProducts> QueryData()
        {
            using (var db = new EcommerceContext())
            {
                // Query our training data from the database. This query is selecting everything from the AdultCensus table. The
                // result is then loaded by ML.Net through the LoadFromEnumerable. LoadFromEnumerable returns an IDataView which
                // can be consumed by an ML.Net pipeline.
                // NOTE: For training, ML.Net requires that the training data is processed in the same order to produce consistent results.
                // Therefore we are sorting the data by the AdultCensusId, which is an auto-generated id.
                // NOTE: That the query used here sets the query tracking behavior to be NoTracking, this is particularly useful because
                // our scenarios only require read-only access.
                foreach (var products in db.FrequentlyBoughtTogetherProducts.AsNoTracking().OrderBy(x => x.ProductID))
                {
                    yield return products;
                }
            }
        }
        public ActionResult Index()
        {
            // Seed the database with the dataset.
            var mlContext = new MLContext(seed: 1);

            /// Query the data from the database, please see <see cref="QueryData"/> for more information.
            var dataView = mlContext.Data.LoadFromEnumerable(QueryData());
            /// Creates the training and testing data sets.
            var trainTestData = mlContext.Data.TrainTestSplit(dataView);
       

            //STEP 3: Your data is already encoded so all you need to do is specify options for MatrxiFactorizationTrainer with a few extra hyperparameters
            //        LossFunction, Alpa, Lambda and a few others like K and C as shown below and call the trainer. 
            MatrixFactorizationTrainer.Options options = new MatrixFactorizationTrainer.Options();
            options.MatrixColumnIndexColumnName = nameof(ProductInfo.ProductID);
            options.MatrixRowIndexColumnName = nameof(ProductInfo.CombinedProductID);
            options.LabelColumnName = "Label";
            options.LossFunction = MatrixFactorizationTrainer.LossFunctionType.SquareLossOneClass;
            options.Alpha = 0.01;
            options.Lambda = 0.025;
         
            options.C = 0.00001;

            //Step 4: Call the MatrixFactorization trainer by passing options.
            var est = mlContext.Recommendation().Trainers.MatrixFactorization(options);

            //STEP 5: Train the model fitting to the DataSet
            //Please add Amazon0302.txt dataset from https://snap.stanford.edu/data/amazon0302.html to Data folder if FileNotFoundException is thrown.
            ITransformer model = est.Fit(trainTestData.TrainSet);
            //STEP 6: Create prediction engine and predict the score for Product 63 being co-purchased with Product 3.
            //        The higher the score the higher the probability for this particular productID being co-purchased 
            var predictionengine = mlContext.Model.CreatePredictionEngine<ProductInfo, ProductPrediction>(model);

            List<(int ProductID, float Score)> deal = new List<(int ProductID, float Score)>();
            List<(int ProductID, float Score)> Top4Deals = new List<(int ProductID, float Score)>();

            var TopDeals = (from m in Enumerable.Range(1, 262111)
                        let p = predictionengine.Predict(
                           new ProductInfo()
                           {
                               ProductID = 3,
                               CombinedProductID = (uint)m
                           })
                        orderby p.Score descending
                        select (ProductID: m, Score: p.Score)).Take(4);
            foreach (var t in TopDeals)
                deal.Add((t.ProductID, t.Score));

            List<t> GetRandomElements<t>(IEnumerable<t> list, int elementsCount)
            {
                return list.OrderBy(x => Guid.NewGuid()).Take(elementsCount).ToList();
            }
            List<(int ProductID, float Score)> randomNumberList = new List<(int ProductID, float Score)>();
            for (int i = 0; i < 4; i++)
            {
                randomNumberList = GetRandomElements(deal, 4);
                foreach (var t in randomNumberList)
                {
                    Top4Deals.Add((t.ProductID, t.Score));
                }
                ViewData["Deal"] = Top4Deals;
            }
            return View();
        }
    }
}

**Comment by luisquintanilla** [comment]
Hi @kamranalam27 

From the code I can't tell what the schema of your input is but most likely the reason why this is happening is the name of your column is something other than label. Your label is the value to predict. Therefore, replace "Label" value with the name of the column containing your value to predict.

See the sample documentation where it shows a `LabelColumnName` other than `Label`. 

https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.recommendationcatalog.recommendationtrainers.matrixfactorization?view=ml-dotnet-preview#Microsoft_ML_RecommendationCatalog_RecommendationTrainers_MatrixFactorization_Microsoft_ML_Trainers_MatrixFactorizationTrainer_Options_

**Comment by muhamedkarajic** [comment]
For me the issue was that the code was trying to create a file while I just gave the folder name which it could not overwrite since its `./` aka root. Therefore `./something` was working.
